RubeesEat/Model/DB/DbPersonRepository.cs

```diff
@@ -41,52 +53,88 @@ public IReadOnlyList<Person> GetAllActive()
             (
                 reader.GetGuid(0),
                 reader.GetString(1),
-                reader.GetString(2)
+                reader.GetString(2),
+                reader.IsDBNull(3) ? null : reader.GetString(3),
+                reader.GetBoolean(4)
+
             ));
         }
 
         return persons.AsReadOnly();
     }
-    
-
 
     public void Add(Person person)
     {
-        using var connection = connectionFactory.CreateDbConnection();
-        using var command = connection.CreateCommand("INSERT INTO Persons (PersonID, FirstName, LastName, IsActive) VALUES (@PersonID, @FirstName, @LastName, @IsActive);");
+        using var connection = _connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand("INSERT INTO Persons (PersonID, FirstName, LastName, LoginName, IsActive) VALUES (@PersonID, @FirstName, @LastName, @LoginName, @IsActive);");
         command.AddParameter("@PersonID", person.Id);
         command.AddParameter("@FirstName", person.FirstName);
         command.AddParameter("@LastName", person.LastName);
+        command.AddParameter("@LoginName", person.LoginName);
         command.AddParameter("@IsActive", person.IsActive);
         
         connection.Open();
         command.ExecuteNonQuery();
     }
     
-    public Person GetCurrentUser()
+    public Person GetOrCreateUser(ClaimsPrincipal user)
     {
-        return GetAll()[^1];
+        var loginName = user.FindFirst("name")?.Value;
+        var nickName = user.FindFirst("nickname")?.Value;
+        var splitNickName = nickName.Split(".");
+        var firstName = splitNickName[0];
+        var lastName = splitNickName[1];
+        if (loginName == null)
+            throw new InvalidOperationException("User is inactive");
```

Comment:
I believe the exception message is not correct. you're talking about a ClaimsPrincipal here. That's an object you got from the authentication infrastructure so there's a user sitting in front of the system. And for some reason, this user did not contain a metadata `name` in its description/bag/whatever-you-call-it. The excpetion should tell you this.

Reply:
This is one of those cases where we discussed something specific with Patrick. The idea is that if the loginName is null, this means (in our application) that the User is inactive. https://re-motion.atlassian.net/jira/software/projects/DSEAT/boards/17?selectedIssue=DSEAT-46

If there is a user sitting in front of the system getting this error message, they should go to an admin and let them know that their user is inactive.

Reply:
I believe there's a misunderstanding. 
Type `User` propety `LoginName` being `null` is exactly what's discussed with Patrick.  
Type `ClaimsPrincipal` not having a `name` entry in its proprties is in my understanding a completely different scenario because it means that the OpenID provider did not send all the fields we expect for the user.

Reply:
I see what you mean. 
I think in this case it's a little strange because we get the loginName from the name entry. We did not know how the loginName would be given in our specific case, when it comes to the company's OpenID provider. Our first guess was that the email address would be the loginName, but for some reason the ClaimsPrincipal did not automatically have the email entry and there was some complication there. So we talked to Patrick and we settled on using the name field for now. Patrick said it didn't matter for the time because we anyhow have to change this once we know exactly which field corresponds to the loginName.

But you mean it should always send us the entries we expect, even when the user is inactive, right? In that case it seems like we would never get the case that we can tell if an inactive user is logging in. If inactive users have no loginName in our db, and such an inactive user were to login, but have a loginName in their ClaimsPrincipal, then the program will just create a new user (since it cannot find the right entry in the db), which is not what we want.

Reply:
> But you mean it should always send us the entries we expect, even when the user is inactive, right? 

More precisely: at this point, the concept of the user being inactive isn't relevant since it's not part of the decision whether to throw an exception. If you want to deny login to inactive users, you would need to check if the user exists and is inactive and if so, then throw a matching exception. For now, as the code stands, we truly only check whether we are getting valid data from the authentiction provider. different subject matter.

>  In that case it seems like we would never get the case that we can tell if an inactive user is logging in.
Correct, see above. might be a new ticket.

>  If inactive users have no loginName in our db, and such an inactive user were to login, but have a loginName in their ClaimsPrincipal, then the program will just create a new user (since it cannot find the right entry in the db), which is not what we want.
True. maybe the concept of having no login-name isn't helpful for inactive users? The problem only exists for legacy data. there, we could simply populate with firstname.lastname@rubicon.eu (with or with out the rubicon suffix) as the login name and then we always have helpful data. Also a separate task.

Either way, this exception right here only valdiates if we get all the data we get from the authentication provider is formatted correctly. if it's not, there is an exception because we can only reject the user. Likely this would be a total system  outtage because all users would be logging in using an invalid data format.

Reply:
I created a new ticket regarding the inactive User problem: https://re-motion.atlassian.net/jira/software/projects/DSEAT/boards/17?selectedIssue=DSEAT-70

I will rename the exception properly, that the claim does not have the proper information